ui/apps/platform/src/Containers/Policies/Wizard/Step3/policyCriteriaDescriptors.tsx

```diff
@@ -272,7 +273,8 @@ export const policyConfigurationDescriptor: Descriptor[] = [
         shortName: 'Not verified by trusted image signers',
         longName: 'Not verified by trusted image signers',
         category: policyCriteriaCategories.IMAGE_REGISTRY,
-        type: 'signaturePolicyCriteria',
+        type: 'tableModal',
+        tableType: 'imageSigning',
```

Comment:
To increase visibility of steps and reduce number of changed files in the future when people imitate this type of policy criteria, is it possible for the descriptor to have a property like `component: ImageSigningTableModal` either instead of or in addition to `tableType` property?

That is, could `PolicyCriteriaFieldInput` instantiate the appropriate type of element directly from descriptor instead of indirectly via `TableModalFieldInput` element?

It would be super cool if TypeScript could enforce in the descriptor that the component has the expected props.

Reply:
On second thought, `Component` uppercase instead of `component` lowercase for key:

```ts
        case 'tableModal': {
            const { Component } = descriptor;
            return (
                <Component
                    setValue={setValue}
                    value={value}
                    readOnly={readOnly}
                />
            );
        }
```

Reply:
Here is first draft of props:

```tsx
type TableModalComponentProps = {
    setValue: (value: unknown) => void;
    value: unknown;
    readOnly?: boolean;
};

type TableModalComponent = (props: TableModalComponentProps) => ReactElement;

export type TableModalDescriptor = {
    type: 'tableModal';
    tableType: string; // TODO is this needed?
    Component: TableModalComponent;
} & BaseDescriptor;
```

Reply:
i'll do this in a separate PR to unblock this one for now.